### DISCUSSION TITLE
[FEATURE] webRTC source pad for gstreamer

### DISCUSSION THREAD

**Discussion by razor123max** [awslabs/amazon-kinesis-video-streams-webrtc-sdk-c#1650] [Ideas]
**Is your feature request related to a problem? Please describe.**
If someone want to use gstreamer pipeline on cloud (ex. ec2 instance), is there anyway we can use kinesis video stream as source in gstremer pipeline?

**Describe the solution you'd like**
Ways to consume kinesis video stream in gstreamer pipline (kinesis -> gstreamer)

**Additional context**
This will greatly help in using Nvidia Deepstream on cloud with individual streams near realtime  inference.

**Comment by hassanctech** [comment]
For sending video to Kinesis Video Streams we have a `kvsink` GStreamer element here: https://github.com/awslabs/amazon-kinesis-video-streams-producer-sdk-cpp

Are you looking for a consumer one `kvssrc` (which does not exist today which consumes from a Kinesis Video Stream?  I'm asking because the issue was opened in WebRTC (which does not actually store anything or even use a Kinesis Video Stream).

Or are you looking for something like this: https://github.com/centricular/webrtcsink

But which communicates with KVS Signaling/STUN/TURN servers (we do not have such an implementation today).

If you can clarify what exactly you're looking for we can mark it as an enhancement.

**Comment by razor123max** [comment]
Thank you for your reply, 
I think i was not clear enough in my request, 

Surely the request is enhancement request, 
This is what i am trying to do,

Edge device Camera --> gstreamer ---> kvsink ............ > **Ec2 instance [kvssrc]** -> gstream -> deepstream

This will enable us to run real-time AI on cloud. 

kvssrc should be able to source webrtc (with kvs signaling) and could be made part pf gstreamer source. then it will be easy to run Nvidia deepstream on cloud. 

or you can point me to any direction where some samples of some version are already implemented? that will be great. 
thanks again.

**Comment by xtianus79** [comment]
Is it possible to do 

`Camera --> gstreamer ---> kvsink ............ `
